src/hiveboard_bridge/src/MessageHandler.cpp

```diff
@@ -31,24 +37,23 @@ bool MessageHandler::handleMessage(MessageDTO message) {
 
                 // Call the right callback
                 if (callback) {
-                    CallbackContext ctx;
-                    ctx.compoundSourceId = message.getSourceId();
-                    ctx.compoundDestinationId = message.getDestinationId();
-                    ctx.moduleDestinationId =
-                        std::get_if<UserCallRequestDTO>(&userCallRequest)->getDestination();
-                    ctx.expectedResponseId = std::get_if<RequestDTO>(&request)->getId();
-
-                    callback.value()(functionArgs, argsLength, ctx);
-                    return true;
+                    callback.value()(functionArgs, argsLength);
+                    responseStatus = GenericResponseStatusDTO::Ok;
                 } else {
+                    responseStatus = GenericResponseStatusDTO::Unknown;
                     ROS_WARN("Function name \"%s\" was not registered as a callback",
                              functionName.c_str());
                 }
             }
         }
     }
-    ROS_WARN("Message not recognised");
-    return false;
+
+    if (responseStatus != GenericResponseStatusDTO::Ok) {
+        ROS_WARN("Message handling failed");
+    }
+
+    return MessageUtils::createResponseMessage(requestId, msgSourceId, msgDestinationId,
+                                               UserCallDestinationDTO::BUZZ, responseStatus, "");
```

Comment:
Destination should not be hardcoded to Buzz. Use the the origin of the message instead.

Reply:
Maybe I'm wrong, but I don't think I have access to that info. The requests messages do not provide the source (BUZZ, HOST, UNKNOWN). In other words, I know when the message is addressed to me, but I don't know who sent it, so I won't be able to address back to the same module. @xgroleau maybe we should look into this.